File: backend/routes/emergency.js
const express = require('express');
const router = express.Router();
const { auth } = require('../middleware/auth');

// SOS Emergency
router.post('/sos', auth, async (req, res) => {
  try {
    const { location, message, emergencyContacts } = req.body;
    // In production: send SMS, push notifications, alert hospitals
    console.log(`🚨 SOS Alert from user ${req.userId}:`, { location, message });

    res.json({
      success: true,
      message: 'Emergency alert sent successfully!',
      alert: {
        id: Date.now().toString(),
        status: 'dispatched',
        nearestHospital: 'AIIMS Delhi',
        estimatedArrival: '8 minutes',
        ambulanceNumber: '108',
        emergencyNumber: '112',
      },
    });
  } catch (error) {
    res.status(500).json({ success: false, message: error.message });
  }
});

// Get emergency contacts
router.get('/contacts', auth, async (req, res) => {
  try {
    const user = req.user;
    res.json({
      success: true,
      contacts: user.health_profile?.emergencyContacts || [],
      defaultNumbers: {
        ambulance: '108',
        emergency: '112',
        poisonControl: '1800-11-6117',
        mentalHealth: '08046110007',
      },
    });
  } catch (error) {
    res.status(500).json({ success: false, message: error.message });
  }
});

module.exports = router;

File: backend/routes/emergency.py
"""
routes/emergency.py — SOS emergency alerts and emergency contacts
Mirrors backend/routes/emergency.js
"""
import time
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel
from typing import Optional, Any

from middleware.auth import get_current_user

router = APIRouter(prefix="/api/emergency", tags=["Emergency"])


class SosRequest(BaseModel):
    location: Optional[Any] = None
    message: Optional[str] = None
    emergencyContacts: Optional[Any] = None


@router.post("/sos")
async def trigger_sos(payload: SosRequest, user: dict = Depends(get_current_user)):
    try:
        print(f"🚨 SOS Alert from user {user.get('id')}: location={payload.location}, msg={payload.message}")
        return {
            "success": True,
            "message": "Emergency alert sent successfully!",
            "alert": {
                "id": str(int(time.time() * 1000)),
                "status": "dispatched",
                "nearestHospital": "AIIMS Delhi",
                "estimatedArrival": "8 minutes",
                "ambulanceNumber": "108",
                "emergencyNumber": "112",
            },
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/contacts")
async def get_emergency_contacts(user: dict = Depends(get_current_user)):
    try:
        health_profile = user.get("health_profile") or {}
        contacts = health_profile.get("emergencyContacts", [])
        return {
            "success": True,
            "contacts": contacts,
            "defaultNumbers": {
                "ambulance": "108",
                "emergency": "112",
                "poisonControl": "1800-11-6117",
                "mentalHealth": "08046110007",
            },
        }
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
